db/model/Gdoc/Gdoc.ts

```diff
@@ -15,12 +15,12 @@ import {
 import { google, Auth, docs_v1 } from "googleapis"
 import { gdocToArchie } from "./gdocToArchie.js"
 import { archieToEnriched } from "./archieToEnriched.js"
-import { createGdocAndInsertOwidArticleContent } from "./archieToGdoc.js"
 
 @Entity("posts_gdocs")
 export class Gdoc extends BaseEntity implements OwidArticleType {
-    @PrimaryColumn() id!: string
-    @Column() slug: string = ""
+    @PrimaryGeneratedColumn() id!: number
+    @Column({ unique: true }) googleId!: string
+    @Column({ unique: true }) slug: string = ""
```

Comment:
I agree that slug should be unique, but that's not how it is currently configured on the database level. Also, we currently have three gdocs posts that don't have a slug set, so... watch out.

Reply:
If I remember correctly, the idea here was to allow drafts to autosave even if authors didn't explicitly set a slug (or a title with auto-slug on). With slugs still coming in from wordpress, setting a unique on slugs in post_gdocs is a good step forward but not foolproof anyway. How does @danyx23's work on slug uniqueness across all resources fit into this picture? I'm a bit fuzzy on the details so it might only apply to redirects? Not entirely sure

Reply:
Yeah, the other problem is that what we have in the TypeORM specification is not authoritative: The table is specified using the migrations that touch `posts_gdocs`, and they don't set a UNIQUE constraint on the slug.
So, currently this is plain wrong, and as we have established it probably _doesn't_ make sense to have this constraint.

Reply:
I'm a bit late to the party but I agree with Marcel. I think the TypeORM should reflect the DB level constraints and on the DB level having slug unique seems a little too strict. We do want unique slugs for published content, but for things registered in the posts_gdocs table I think it should be allowed to have conflicting/missing slugs (they should be surfaced in the UI and at previewing/publishing time of course).